web/src/main/java/org/cbioportal/web/ClinicalDataController.java

```diff
@@ -13,54 +24,125 @@
 import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
+import java.util.ArrayList;
 import java.util.List;
 
 @RestController
 @Api(tags = "Clinical Data", description = " ")
 public class ClinicalDataController {
 
-    @RequestMapping(value = "/studies/{studyId}/samples/{sampleId}/clinical-data", method = RequestMethod.GET)
-    @ApiOperation("Get all clinical data in a sample in a study")
-    public ResponseEntity<List<? extends ClinicalDataSummary>> getAllClinicalDataInSampleInStudy(@PathVariable String studyId,
-                                                                                                 @PathVariable String sampleId,
-                                                                                                 @RequestParam(required = false) String attributeId,
-                                                                                                 @RequestParam(defaultValue = "SUMMARY") Projection projection,
-                                                                                                 @RequestParam(defaultValue = PagingConstants.DEFAULT_PAGE_SIZE) Integer pageSize,
-                                                                                                 @RequestParam(defaultValue = PagingConstants.DEFAULT_PAGE_NUMBER) Integer pageNumber) {
+    @Autowired
+    private ClinicalDataService clinicalDataService;
 
-        throw new UnsupportedOperationException();
+    @RequestMapping(value = "/studies/{studyId}/samples/{sampleId}/clinical-data", method = RequestMethod.GET,
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiOperation("Get all clinical data of a sample in a study")
+    public ResponseEntity<List<? extends ClinicalDataSummary>> getAllClinicalDataOfSampleInStudy(
+            @PathVariable String studyId,
+            @PathVariable String sampleId,
+            @RequestParam(required = false) String attributeId,
+            @RequestParam(defaultValue = "SUMMARY") Projection projection,
+            @RequestParam(defaultValue = PagingConstants.DEFAULT_PAGE_SIZE) Integer pageSize,
```

Comment:
These parameters could benefit from @ApiParam human-understandable explanations of the behavior of arguments. (Unless they are explained in the overall ApiOperation comment). For instance, if there is an absolute upper limit on the user specifiable page size, it could be commented on here, as well as what will happen on an out-of-range error (parameter forced in to bounds .. or error code ... or null result?)
Adding these comments could be a future improvement.